### PR TITLE
Generate a demo for new icons when updating icons

### PR DESCRIPTION
**Pull request by guidezpl**
This makes generating screenshots for new icons easier.

### CODE REVIEW

File: dev/tools/update_icons.dart [hunk 1]
```diff
@@ -450,6 +492,8 @@ class Icon {
   String get dartDoc =>
       '<i class="material-icons$htmlSuffix md-36">$shortId</i> &#x2014; $family icon named "$name"$style';
```
Comment (clocksmith) [inline dev/tools/update_icons.dart:493]: nit: not sure if there should be space between name and style

**Review by clocksmith — APPROVED** [review]
cool, LGTM

File: dev/tools/test/update_icons_test.dart [hunk 2]
```diff
@@ -40,4 +40,15 @@ void main() {
   test('no double underscores', () {
     expect(Icon.generateFlutterId('abc__123'), 'abc_123');
   });
+
+  test('usage is correct', () {
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply dev/tools/test/update_icons_test.dart:44]: updated

File: dev/tools/update_icons.dart (hunk 1 above)
Comment (guidezpl) [reply dev/tools/update_icons.dart:493]: the space is already included, all good

File: dev/tools/update_icons.dart [hunk 3]
```diff
@@ -379,18 +385,54 @@ bool testIsStable(Map<String, String> newCodepoints, Map<String, String> oldCode
...
+void _generateIconDemo(File demoFilePath, Map<String, String> tokenPairMap) {
+  if (tokenPairMap.isEmpty) {
+    stderr.writeln('No new icons, skipping generating icon demo');
+    return;
+  }
+  stderr.writeln('Generating icon demo at $_defaultDemoFilePath');
+
+  final StringBuffer newIconUsages = StringBuffer();
+  for (final MapEntry<String, String> entry in tokenPairMap.entries) {
+    newIconUsages.writeln(Icon(entry).usage);
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply dev/tools/update_icons.dart:405]: just a temp buffer that's used to construct `demoFileContents` below

**Pull request merged**